Tasks/Common/AzureRmDeploy-common/azure-arm-rest/webClient.ts

```diff
@@ -89,7 +89,7 @@ export function sleepFor(sleepDurationInSeconds): Promise<any> {
 }
 
 async function sendRequestInternal(request: WebRequest): Promise<WebResponse> {
-    tl.debug(util.format("[%s]%s", request.method, request.uri));
+    tl.debug(util.format("[%s]%s", request.method, maskCredentialsFromUrl(request.uri)));
```

Comment:
this could still be leaky. someone can print uri somewhere else without masking. Is there another way to provide auth other than embedding it in uri?

Reply:
won't setting it as a auth header of type basic work?

Reply:
in case no other solution exists, we should not make auth part of uri property. rather do directly when calling into `httpCallbackClient.request`

some thing like 
```
httpCallbackClient.request(request.method, addAuthToUri(request.uri, credentials), request.body, request.headers)
```